### PR TITLE
Pass the entire slice to selectors function

### PR DESCRIPTION
**Pull request by robbaman**
Changed the selector definition to be the entire signalSlice as oposed to just the state.

Not sure how to run the tests (the docs just say to 'run the tests'), so not sure if this also required a change to the test framework. The changed version did do what I'd expect in my project.

### CODE REVIEW

**Comment by joshuamorony** [comment]
Note to reviewers this was discussed/agreed in this issue: https://github.com/nartc/ngxtension-platform/issues/187

@robbaman you can run the tests with `pnpm exec nx run ngxtension/signal-slice:test`, this likely will not break any existing tests but ideally it would be good to add an extra test where one of the top level selectors is accessed to create a custom selector

**Comment by joshuamorony** [comment]
You should be able to solve this by passing in `any` instead of `TSelectors` when typing the `selectors` config:

```ts
	selectors?: (
		state: SignalSlice<
			TSignalValue,
			TActionSources,
			any,
			TEffects,
			TActionEffects
		>
	) => TSelectors;
```

File: libs/ngxtension/signal-slice/src/signal-slice.ts [hunk 1]
```diff
@@ -140,7 +140,15 @@ export function signalSlice<
...
-	selectors?: (state: Signal<TSignalValue>) => TSelectors;
+	selectors?: (
+		state: SignalSlice<
+			TSignalValue,
+			TActionSources,
+			TSelectors,
+			any,
+			TActionEffects
+		>
+	) => TSelectors;
```
[the remark this answers is not in the document]
Comment (robbaman) [reply libs/ngxtension/signal-slice/src/signal-slice.ts]: You're correct. I've made the change and also changed the test file to read the `age` property from `state().age` to `state.age()` to confirm it is indeed the entire slice that is provided.

**Review by joshuamorony — APPROVED** [review]
LGTM, thanks!

Will just need to wait on a reviewer with write access to trigger the workflow and get this merged

**Review by nartc — APPROVED** [review]

**Comment by nartc** [comment]
@all-contributors please add @robbaman for code

**Pull request merged**